src/whylogs/core/statistics/stringtracker.py

```diff
@@ -0,0 +1,245 @@
+
+        self.char_pos_map = [NumberTracker() for _ in self.character_list]
+
+    def update(self, value):
+
+        for indx, char in enumerate(value):
+            if char in self.character_list:
+                self.char_pos_map[self._character_list_map[char]].track(indx)
+
+    def merge(self, other):
+        """
+        Merges two Char Pos Frequency Maps
+
+        Args:
+
+            other (CharPosTracker): to be merged
+
+        """
+        new_character_list = list(set(self.character_list + other.character_list))
+
+        # initialize merged
+        new_char_pos_tracker = CharPosTracker(character_list=new_character_list)
+
+        # merge
+        new_indxes = encode_to_integers(self.character_list, new_character_list)
+
+        new_char_pos_tracker = new_char_pos_tracker.char_pos_map
+
+        for old_indx, new_indx in enumerate(new_indxes):
+            new_char_pos_tracker[new_indx].merge(self.char_pos_map[old_indx])
+
+        new_indxes = encode_to_integers(other.character_list, new_character_list)
+
+        for old_indx, new_indx in enumerate(new_indxes):
+            new_char_pos_tracker[new_indx].merge(other.char_pos_map[old_indx])
+
+        return new_char_pos_tracker
+
+    def to_protobuf(self):
+        """
+        Return the object serialized as a protobuf message
+        """
+        opts = dict(
+            char_list=self.character_list,
+            char_pos_map=[nt.to_protobuf() for nt in self.char_pos_map],
+        )
+
+        msg = CharPosMessage(**opts)
+        return msg
+
+    @staticmethod
+    def from_protobuf(self, message: CharPosMessage):
+        """
+        Load from a protobuf message
+
+        Returns
+        -------
+        CharPosTracker
+        """
+
+        opts = dict(
+            character_list=CharPosMessage.char_list,
+        )
+        char_pos_tracker = CharPosTracker(**opts)
+        char_pos_tracker.char_pos_map = [nt.from_protobuf for nt in CharPosMessage.char_pos_map]
+        return char_pos_tracker
+
+    def to_summary(
+        self,
+    ):
+        opts = dict(character_list=self.character_list, char_pos_map=[nt.to_summary() for nt in self.char_pos_map])
+
+        return CharPosSummary(**opts)
+
+
+class StringTracker:
+    """
+    Track statistics for strings
+
+    Parameters
+    ----------
+    count : int
+        Total number of processed values
+    items : frequent_strings_sketch
+        Sketch for tracking string counts
+    theta_sketch : ThetaSketch
+        Sketch for approximate cardinality tracking
+    length : NumberTracker
+        tracks the distribution of length of strings
+    token_length
+    """
+
+    def __init__(
+        self,
+        count: int = None,
+        items: frequent_strings_sketch = None,
+        theta_sketch: ThetaSketch = None,
+        length: NumberTracker = None,
+        token_length: NumberTracker = None,
+        char_pos_tracker=None,
```

Comment:
spacing (styling)

Reply:
Will double check, seems our format/fix missed this and all the protobuf, will look into it.

Reply:
I don't think the formatter looks at protobuf yet btw!